shumai/cpp/flashlight_binding.cc

```diff
@@ -90,6 +95,34 @@ JSTypedArrayBytesDeallocator genTensorDestroyer() {
   return destroyTensor;
 }
 
+// `grad_in` is Shumai equivalent for Flashlight `gradOutput`
+void* conv2dBackwardData(void* grad_in,
+                         void* in,
+                         void* wt,
+                         void* bs,
+                         int sx,
+                         int sy,
+                         int px,
+                         int py,
+                         int dx,
+                         int dy,
+                         int groups) {
+  LOCK_GUARD
+  auto* used_grad_in = reinterpret_cast<fl::Tensor*>(grad_in);
+  auto* used_bs = reinterpret_cast<fl::Tensor*>(bs);
+  auto* used_in = reinterpret_cast<fl::Tensor*>(in);
+  auto* used_wt = reinterpret_cast<fl::Tensor*>(wt);
+
+  auto payload = std::make_shared<fl::detail::AutogradPayload>();
+  std::shared_ptr<fl::DynamicBenchmark> dataBench;
+  auto result =
+      used_in->backend()
+          .getExtension<fl::AutogradExtension>()
+          .conv2dBackwardData(*used_grad_in, *used_in, *used_wt, sx, sy, px, py,
+                              dx, dy, groups, dataBench, payload);
+  return new fl::Tensor(result);
```

Comment:
before returning, always update the byte tracker `g_bytes_used += result.bytes();`

Reply:
I really should set up a convention here... e.g. `RETURN_TENSOR(result)` and that does all the logic we need (like `LOCK_GUARD`)

Reply:
Yea, that convention would probably keep the code a bit cleaner, so I'm a proponent of adding `RETURN_TENSOR`!